app/services/EssttpBarsService.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import essttp.rootmodel.bank.{BankDetails, TypeOfBankAccount, TypesOfBankAccount}
+import models.bars.BarsCommon.{BarsBankAccount, BarsResponse, BarsTypeOfBankAccount, BarsTypesOfBankAccount}
+import models.bars.BarsVerifyRequest._
+import models.enumsforforms.IsSoleSignatoryFormValue
+import play.api.mvc.RequestHeader
+import services.EssttpBarsService._
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+ * ESSTTP-specific interface to Bank Account Reputation service (BARs).
+ */
+@Singleton
+class EssttpBarsService @Inject() (barsService: BarsService)(implicit ec: ExecutionContext) {
+
+  def validateBankAccount(bankDetails: BankDetails)(implicit requestHeader: RequestHeader): Future[BarsResponse] =
+    barsService.validateBankAccount(toBarsBankAccount(bankDetails))
+
+  def verifyPersonal(bankDetails: BankDetails)(implicit requestHeader: RequestHeader): Future[BarsResponse] =
+    barsService.verifyPersonal(toBarsBankAccount(bankDetails), toBarsSubject(bankDetails))
+
+  def verifyBusiness(bankDetails: BankDetails)(implicit requestHeader: RequestHeader): Future[BarsResponse] =
+    barsService.verifyBusiness(toBarsBankAccount(bankDetails), Some(toBarsBusiness(bankDetails)))
+
+  // TODO ItSpec
+  def verifyBankDetails(
+      bankDetails:       BankDetails,
+      isSoleSignatory:   IsSoleSignatoryFormValue,
+      typeOfBankAccount: TypeOfBankAccount
+  )(implicit requestHeader: RequestHeader): Future[Option[BarsResponse]] = {
```

Comment:
it's a bit annoying that the `nameMatches` field and some other fields can be `None` even when we get a `validateFailure` response from the validate bank details form cos we're gonna have to code around a case that shouldn't exist. There are couple of ways to avoid that, one of which using the type of the response to give you more info like:
```scala
sealed trait BarsResponse
case class InvalidBankDetails(response: BarsValidateResponse) extends BarsResponse
case class ValidBankDetails(response: BarsVerifyResponse) extends BarsResponse
```
There may be a nicer way than that.